### PR TITLE
fix #1153 Ensure only Content-Length is set when creating full response

### LINKED ISSUE #1153: Transfer-Encoding & Content-Length headers are being sent in response headers.
When a string is sent in the response body, `GET` request has only `Content-Length` whereas  `HEAD` request has both `Transfer-Encoding` and `Content-Length`.

## Expected Behavior
Should remove `Transfer-Encoding` header for HEAD request too, as it is being done for `GET` request.

## Actual Behavior
Both `Transfer-Encoding` and `Content-Length`  are being sent.
### Worth noting
[https://tools.ietf.org/html/rfc7230#section-3.3.2](https://tools.ietf.org/html/rfc7230#section-3.3.2) 
> A sender MUST NOT send a Content-Length header field in any message
   that contains a Transfer-Encoding header field.

## Steps to Reproduce
* Controller code
```java
@GetMapping("/message")
public ResponseEntity<String> getMessage() {
    return  ResponseEntity.ok("foo");
}
```
* Result of GET API
```
curl -v -X GET http://localhost:8080/message
Note: Unnecessary use of -X or --request, GET is already inferred.
*   Trying ::1...
* TCP_NODELAY set
* Connected to localhost (::1) port 8080 (#0)
> GET /message HTTP/1.1
> Host: localhost:8080
> User-Agent: curl/7.64.1
> Accept: */*
>
< HTTP/1.1 200 OK
< Content-Type: text/plain;charset=UTF-8
< Content-Length: 3
<
* Connection #0 to host localhost left intact
foo* Closing connection 0
```
* Result of HEAD API
```
 curl -v -X HEAD http://localhost:8080/message
Warning: Setting custom HTTP method to HEAD with -X/--request may not work the
Warning: way you want. Consider using -I/--head instead.
*   Trying ::1...
* TCP_NODELAY set
* Connected to localhost (::1) port 8080 (#0)
> HEAD /message HTTP/1.1
> Host: localhost:8080
> User-Agent: curl/7.64.1
> Accept: */*
>
< HTTP/1.1 200 OK
< transfer-encoding: chunked
< Content-Type: text/plain;charset=UTF-8
< Content-Length: 3
<
```

## Possible Solution
https://github.com/reactor/reactor-netty/blob/abd6ac4e2e8dd2d129929f9451e7276903d53564/src/main/java/reactor/netty/http/server/HttpServerOperations.java#L161
has a condition to remove transfer-encoding only if it is not a HEAD request.

## Your Environment
Environment:
spring-boot:  2.2.6.RELEASE
Using spring-boot-starter-webflux with Netty.

### PR DESCRIPTION
**Pull request by violetagg**
Fixes #1153

### CODE REVIEW

File: src/test/java/reactor/netty/http/server/HttpServerTests.java [hunk 1]
```diff
@@ -554,38 +558,95 @@ public void testContentLengthHeadRequest() {
...
+		doTestContentLengthHeadRequest("/8", disposableServer.address(), HttpMethod.HEAD, sentHeaders, false, false);
+		doTestContentLengthHeadRequest("/9", disposableServer.address(), HttpMethod.HEAD, sentHeaders, false, false);
+		doTestContentLengthHeadRequest("/10", disposableServer.address(), HttpMethod.HEAD, sentHeaders, false, false);
+		doTestContentLengthHeadRequest("/11", disposableServer.address(), HttpMethod.HEAD, sentHeaders, false, false);
+		doTestContentLengthHeadRequest("/12", disposableServer.address(), HttpMethod.HEAD, sentHeaders, false, false);
+		doTestContentLengthHeadRequest("/13", disposableServer.address(), HttpMethod.HEAD, sentHeaders, false, false);
+	}
+
+	@Test
+	public void testIssue() {
```
Comment (simonbasle) [inline src/test/java/reactor/netty/http/server/HttpServerTests.java:632]: that would benefit from a more descriptive name :)

Comment (violetagg) [reply src/test/java/reactor/netty/http/server/HttpServerTests.java:632]: oppps :)

File: src/main/java/reactor/netty/http/server/HttpServerOperations.java [hunk 2]
```diff
@@ -158,13 +158,11 @@ protected HttpMessage newFullBodyMessage(ByteBuf body) {
 		HttpResponse res =
 				new DefaultFullHttpResponse(version(), status(), body);
 
-		if (!HttpMethod.HEAD.equals(method())) {
```
Comment (kaladhar-mummadi) [inline src/main/java/reactor/netty/http/server/HttpServerOperations.java:161]: @violetagg , do you have a context of why this condition was written in the first place?

Comment (violetagg) [reply src/main/java/reactor/netty/http/server/HttpServerOperations.java:161]: it is just a bug

**Comment by violetagg** [comment]
@mrbrown-09 The release is planned for Monday 20.07

**Comment by mrbrown-09** [comment]
great, ty very much!

**Pull request merged**